libbeat/common/mapstr.go

```diff
@@ -102,9 +102,17 @@ func deepUpdateValue(old interface{}, val MapStr, overwrite bool) interface{} {
 
 	switch sub := old.(type) {
 	case MapStr:
+		if sub == nil {
+			return val
+		}
```

Comment:
Surprised that this isn't caught already by the `if old == nil` check at the start of this function. 🤔

Reply:
Just found https://groups.google.com/forum/#!topic/golang-nuts/wnH302gBa4I/discussion. 

Should we remove the code on lines 99-101?

Reply:
There is `nil`, and then there is `nil` :)

An `interface` value is a tuple `(type, value)`. An `interface` is nil, if and only if both `type` and `value` are `nil`. This is why the initial check fails here. We have a type `(type MapStr, nil)`. The `switch` statement only looks for `type`, unpacks it, and we finally have the `nil` value in `sub`. This is when the call panics.

Removing line 99 should be safe. E.g.: https://play.golang.org/p/uajkDxQW9qH